### PR TITLE
add support for []int

### PR DESCRIPTION
**Pull request by marco709394**
Encode int element as int64

### CODE REVIEW

**Comment by ineiti** [comment]
So this can 'fail' if the sender wants to send a 64-bit int and the receiver only has 32-bit int. It will probably just ignore the too-big value. But this might have consequences.

- either we don't allow `[]int`-encoding (but then we should take away the `int`, too)
- we write a test on the receiver side to see if the int is too big and fail if it is
- we ignore it and let programmers deal with it
- we give out a warning if `protobuf.INTS_ARE_BAD` is not set to 0 and we're encountering an `int` or `[]int`.

**Comment by Liamsi** [comment]
> So this can 'fail' if the sender wants to send a 64-bit int and the receiver only has 32-bit int. It will probably just ignore the too-big value.

Are sure? On first sight it doesn't seem so.

File: encode.go [hunk 1]
```diff
@@ -287,6 +287,11 @@ func (en *encoder) slice(key uint64, slval reflect.Value) {
...
 		}
 
+	case []int:
+		for i := 0; i < sllen; i++ {
+			packed.svarint(int64(slt[i]))
+		}
+
 	case []int32:
 		for i := 0; i < sllen; i++ {
 			packed.svarint(int64(slt[i]))
```
Comment (nikkolasg) [inline encode.go]: Mhh i was not aware we use `[]int64` when we want to send `[]int32`, maybe we should change that too to stay consistent so developers can still use this library even if it only supports 32 bits integers ?

**Comment by Liamsi** [comment]
I mean, `int` will always be encoded as `int64` here or not: https://github.com/dedis/protobuf/pull/28/files#diff-e214241880c7c5dc058ccee5f600a351R290

**Comment by Liamsi** [comment]
OK, I see the problem you described above @ineiti ...

File: encode.go (hunk 1 above)
Comment (ineiti) [inline encode.go]: How does the official protobuf do it?

Comment (marco709394) [inline encode.go]: [https://github.com/golang/protobuf/blob/master/proto/encode.go#L732](url)
I am not so sure about the meaning of conversion to uint64 here, you guys may be more familiar with it :)

Comment (Liamsi) [inline encode.go]: > How does the official protobuf do it?

In original protobuf you basically forced to choose: there is no `int` in the .proto language definition: https://developers.google.com/protocol-buffers/docs/proto#scalar (<- see .proto type vs Go type)

The most consistent and reasonable approach would be to simply not allow golang's `int`. But if we want to allow that we should always make sure that on the receiving side there can't be an overflow and else panic.

**Comment by Liamsi** [comment]
My opinion might be buried in this comment: https://github.com/dedis/protobuf/pull/28#discussion_r70075003

**Comment by ineiti** [comment]
For simplicity, just use the following:

`we write a test on the receiver side to see if the int is too big and fail if it is`

**Comment by ineiti** [comment]
Done in #42

**Pull request closed without merging**